### PR TITLE
Fixes Disappearing player drops during pvp, and protects decorated pots on 1.20.3/4

### PR DESCRIPTION
**Pull request by QarthO**
In 1.20.3 the decorated pot can now be interacted with to store items, and can be destroyed by being hit with projectiles, (similarly to the chorus fruit).

These changes add the decorated pot to the list of blocks to check when interacting, and upon being shot by a projectile.

### CODE REVIEW

**Comment by Jikoo** [comment]
Should also bump api-version to 1.20. The only reason it wasn't bumped when the API was updated before was because only an event was used, and Bukkit handles those not being present pretty gracefully.

**Comment by QarthO** [comment]
Are we sure? https://docs.papermc.io/paper/dev/plugin-yml#api-version
image
If we bump the api-version to 1.20, then the plugin will no longer work on paper servers 1.19 and below.

**Comment by Jikoo** [comment]
It's currently 1.17, have pots really been around that long? I guess they might have been feature preview in 1.19, so maybe that's okay.

**Comment by QarthO** [comment]
good point, realistically we can just drop support, since the older versions of gp are perfectly fine

**Review by nouish — COMMENTED** [review]
Ref:
```java
 private void chorusFlower(ProjectileHitEvent event)
 ```

Might be a good idea to rename this now that the event is used for more than chorus flowers. Even just `onProjectileHit` maybe?

**Comment by JerichoTorrent** [comment]
Glad to see my suggestion is being considered:D GP is the best.

**Comment by QarthO** [comment]
Didn't mean to include in it in this pull...but just fixed an issue where pvp deaths result in players not dropping their items (started from Paper builds 315 and later)

This paper pull: https://github.com/PaperMC/Paper/pull/7650 causes the issues.

The issue is that the event ``PlayerDropItemEvent`` is now being fired when the player dies (including during pvp) on all of the item stacks in the player's inventory. Since we were listening to this event to prevent players from dropping items during pvp, (the event gets canceled) the player's items aren't dropped. If I had to guess, the player's inventory is cleared after this, so their items are never dropped and then deleted. Thankfully the fix is just to check if the player is dead when listening to this event.

I'm not sure if it was intentional on paper to make this change, it is weird that ``PlayerDropItemEvent`` is only fired during a player death. Regardless of whether paper backtracks, this change will fix the issue, and won't have any issues going forward

**Comment by SergioMol98** [comment]
Hi QarthO, im new here and wanna know how can i implements this changes in my server, i have the problem that players disconect during pvp and die but they are not dropping items, i just have to download again the latest version?

Best

**Comment by QarthO** [comment]
I'm lazy and just going to copy and paste what I put in the discord

# Warning: Updating Paper 1.20.2 to builds 315 to 333 can delete items during pvp

:white_check_mark:  **Update: Starting in Paper 1.20.4 build __334__ this has been fixed**

TLDR: Paper made a change 2 weeks ago, and now if your using griefprevention with "``PvP.AllowCombatItemDrop``" set to false. Whenever someone is killed all of their items will be deleted, and not dropped

To prevent it:
- Don't update Paper yet
> Only builds 315 and later have the issue. I'd recommend staying on 314 and either wait until paper reverts the change, or GriefPrevention gets updated
> It's ok to downgrade to build 314 ONLY if you're staying on the same mc-version

If you want to update Paper before GP officially updates do one of the following:
- Set ``PvP.AllowCombatItemDrop`` to true
> You can update paper and it should be fine
- OR use the dev build from my PR
> Pull Request: https://github.com/TechFortress/GriefPrevention/pull/2181
> Download Dev Build: https://ci.appveyor.com/project/RoboMWM39862/griefprevention/builds/48778704/artifacts
> *Keep in mind these are Dev/Experimental builds

If you'd like to know specifics on what/why it broke
> https://github.com/TechFortress/GriefPrevention/pull/2181#issuecomment-1859553179

I also made an issue on Paper discussing about the change
> https://github.com/PaperMC/Paper/issues/10044

**Comment by SrBedrock** [comment]
> I'm lazy and just going to copy and paste what I put in the discord
> 
> # Warning: Updating Paper 1.20.2 to builds 315 to 333 can delete items during pvp
> ✅ **Update: Starting in Paper 1.20.4 build 334 this has been fixed**
> 
> TLDR: Paper made a change 2 weeks ago, and now if your using griefprevention with "`PvP.AllowCombatItemDrop`" set to false. Whenever someone is killed all of their items will be deleted, and not dropped
> 
> To prevent it:
> 
> * Don't update Paper yet
> 
> > Only builds 315 and later have the issue. I'd recommend staying on 314 and either wait until paper reverts the change, or GriefPrevention gets updated
> > It's ok to downgrade to build 314 ONLY if you're staying on the same mc-version
> 
> If you want to update Paper before GP officially updates do one of the following:
> 
> * Set `PvP.AllowCombatItemDrop` to true
> 
> > You can update paper and it should be fine
> 
> * OR use the dev build from my PR
> 
> > Pull Request: #2181
> > Download Dev Build: [ci.appveyor.com/project/RoboMWM39862/griefprevention/builds/48778704/artifacts](https://ci.appveyor.com/project/RoboMWM39862/griefprevention/builds/48778704/artifacts)
> > *Keep in mind these are Dev/Experimental builds
> 
> If you'd like to know specifics on what/why it broke
> 
> > [#2181 (comment)](https://github.com/TechFortress/GriefPrevention/pull/2181#issuecomment-1859553179)
> 
> I also made an issue on Paper discussing about the change
> 
> > [PaperMC/Paper#10044](https://github.com/PaperMC/Paper/issues/10044)

I believe it has already been resolved by the Paper.

**Comment by QarthO** [comment]
> > I'm lazy and just going to copy and paste what I put in the discord
> > # Warning: Updating Paper 1.20.2 to builds 315 to 333 can delete items during pvp
> > ✅ **Update: Starting in Paper 1.20.4 build 334 this has been fixed**
> > TLDR: Paper made a change 2 weeks ago, and now if your using griefprevention with "`PvP.AllowCombatItemDrop`" set to false. Whenever someone is killed all of their items will be deleted, and not dropped
> > To prevent it:
> > 
> > * Don't update Paper yet
> > 
> > > Only builds 315 and later have the issue. I'd recommend staying on 314 and either wait until paper reverts the change, or GriefPrevention gets updated
> > > It's ok to downgrade to build 314 ONLY if you're staying on the same mc-version
> > 
> > 
> > If you want to update Paper before GP officially updates do one of the following:
> > 
> > * Set `PvP.AllowCombatItemDrop` to true
> > 
> > > You can update paper and it should be fine
> > 
> > 
> > 
> > * OR use the dev build from my PR
> > 
> > > Pull Request: #2181
> > > Download Dev Build: [ci.appveyor.com/project/RoboMWM39862/griefprevention/builds/48778704/artifacts](https://ci.appveyor.com/project/RoboMWM39862/griefprevention/builds/48778704/artifacts)
> > > *Keep in mind these are Dev/Experimental builds
> > 
> > 
> > If you'd like to know specifics on what/why it broke
> > > [#2181 (comment)](https://github.com/TechFortress/GriefPrevention/pull/2181#issuecomment-1859553179)
> > 
> > 
> > I also made an issue on Paper discussing about the change
> > > [PaperMC/Paper#10044](https://github.com/PaperMC/Paper/issues/10044)
> 
> I believe it has already been resolved by the Paper.

Yes. Read the post, I said it was already fixed by paper :)

**Comment by RoboMWM** [comment]
Ideally would like this to be two PRs since this addresses two entirely different things. I'll pull tho for now..

**Review by RoboMWM — APPROVED** [review]

**Comment by QarthO** [comment]
> Ideally would like this to be two PRs since this addresses two entirely different things. I'll pull tho for now..

yeah, wasnt intentional. forgot to switch branches and accidentally committed to this pr. after i found out was too lazy to fix it
